diff --git a/extensions/pyo3/private/pyo3.bzl b/extensions/pyo3/private/pyo3.bzl
index 8da97358..d0e5c0a8 100644
--- a/extensions/pyo3/private/pyo3.bzl
+++ b/extensions/pyo3/private/pyo3.bzl
@@ -250,6 +250,7 @@ def pyo3_extension(
         data = [],
         deps = [],
         edition = None,
+        lint_config = None,
         imports = [],
         proc_macro_deps = [],
         rustc_env = {},
@@ -277,6 +278,8 @@ def pyo3_extension(
         edition (str, optional): The rust edition to use for this crate. Defaults to the edition specified in the rust_toolchain.
             For more details see [rust_shared_library][rsl].
+        lint_config (Label, optional): A target providing the lint configuration for the Rust shared library.
+            For more details see [rust_shared_library][rsl].
         imports (list, optional): List of import directories to be added to the `PYTHONPATH`.
             For more details see [py_library.imports][pli].
         proc_macro_deps (list, optional): List of `rust_proc_macro` targets used to help build the library target.
@@ -330,6 +333,7 @@ def pyo3_extension(
         deps = [
             Label("//private:current_rust_pyo3_toolchain"),
             Label("@rules_python//python/cc:current_py_cc_headers"),
         ] + deps,
+        lint_config = lint_config,
         edition = edition,
         proc_macro_deps = proc_macro_deps,
         rustc_env = rustc_env,
